annet/gen.py

```diff
@@ -232,11 +233,30 @@ def _old_new_per_device(ctx: OldNewDeviceContext, device: Device, filterer: Filt
 
         entire_results = res.entire_results
         json_fragment_results = res.json_fragment_results
+        old_json_fragment_files = old_files.json_fragment_files
```

Comment:
Down the line we are essentially assume implementation of `_update_json_config`, being that it modifies the `old_json_fragment_files` dict inplace, since:

1) we make a new reference `old_json_fragment_files = old_files.json_fragment_files`
2) pass it `old_json_fragment_files = _update_json_config(old_json_fragment_files)`
3) return original reference `OldNewResult(old_json_fragment_files=old_files.json_fragment_files)`

This is confusing and error-prone. We should either use `old_json_fragment_files` in return or not create the new reference at all.